src/main/java/org/folio/rest/impl/Processor.java

```diff
@@ -429,16 +442,16 @@ private String processRules(String data, JsonArray rules, Leader leader){
     //there are rules associated with this subfield / control field - to instance field mapping
     String originalData = data;
     for (int ruleIndex = 0; ruleIndex < rules.size(); ruleIndex++) {
-      ProcessedRule pr = processRule(rules.getJsonObject(ruleIndex), leader, data, originalData);
-      data = pr.getData();
-      if (pr.doBreak()) {
+      ProcessedSingleItem psi = processRule(rules.getJsonObject(ruleIndex), leader, data, originalData);
+      data = psi.getData();
+      if (psi.doBreak()) {
```

Comment:
Wrapping data and doBreak in ProcessedSingleItem is more complicated than directly returning data and using `java.util.concurrent.atomic.AtomicBoolean doBreak`:
```
data = processRule(rules.getJsonObject(ruleIndex), leader, data, originalData, doBreak);
if (doBreak.get()) {
  break;
}
```